### ISSUE TITLE
Since 1.197, with Net::DNS installed, mx no longer passes A records

### ISSUE THREAD

**Issue by dracos** [Perl-Email-Project/Email-Valid#30]
With Email::Valid 1.196 and Net::DNS:
```
$ perl -e'use Email::Valid; $em = Email::Valid->new(-mxcheck => 1); my $f = $em->address("test\@newswww.bbc.net.uk"); print "$f\n"'
test@newswww.bbc.net.uk
```
With Email::Valid 1.198 and Net::DNS:
```
$ perl -e'use Email::Valid; $em = Email::Valid->new(-mxcheck => 1); my $f = $em->address("test\@newswww.bbc.net.uk"); print "$f\n"'

```

The issue is caused by the change to using Net::DNS::mx in 1.197 #18, which only searches MXs. If Email::Valid is no longer going to allow As without MXs, then the documentation of `mx()` needs to be fixed, as the code comment still says "# Purpose: Check whether a DNS record (A or MX) exists for a domain." and the pod says "This method accepts an email address or domain name and determines whether a DNS record (A or MX) exists for it."

If Net::DNS is not installed, then As are passed as valid.

**Comment by rjbs** [comment]
Thanks, in retrospect #18 is puzzling.  I'll look into this.

**Comment by tmorehouse** [comment]
A domain without MX records typically cannot receive email.  When I do a DNS query for MX records for `newswww.bbc.net.uk` I get no answer:

        troy@lennier:/# nslookup
        > set type=MX
        > newswww.bbc.net.uk
        Server 8.8.8.8
        Address: 8.8.8.8#53
        
        Non-authoritative answer:
        
        *** Can't find newswww.bbc.net.uk: No answer

**Comment by dracos** [comment]
@tmorehouse The spec says they can, https://tools.ietf.org/html/rfc5321#section-5.1 is quite clear: "If an empty list of MXs is returned, the address is treated as if it was associated with an implicit MX RR, with a preference of 0, pointing to that host." – there is no way of checking deliverability apart from trying to deliver, as Email::Valid says in its documentation. Of course most don't, but fundamentally that's not my point – the issue is `mx` *used* to do this, and the docs still says it *does* do this, and if you don't have Net::DNS installed it does do this, so this is an inconsistent breaking change in the functionality. I don't care whether it does or not, I'd be happy if it only checked MX or didn't, but I do care that either the code changes back or the documentation changes to match.

**Comment by rjbs** [comment]
Fixed by #31, but also created #33.

**Comment by dracos** [comment]
Thanks :) Don't suppose you have time to take a look at https://github.com/Perl-Email-Project/Email-Address/pull/14 while you're here? :)
